src/ConnectionAutomaton.cpp

```diff
@@ -98,9 +101,11 @@ void ConnectionAutomaton::closeDuplexConnection(folly::exception_wrapper ex) {
 }
 
 void ConnectionAutomaton::reconnect(
-    std::unique_ptr<DuplexConnection> newConnection) {
+  std::unique_ptr<DuplexConnection> newConnection,
+  std::unique_ptr<ClientResumeStatusCallback> statusCallback) {
   disconnect();
   connection_ = std::shared_ptr<DuplexConnection>(std::move(newConnection));
+  resumeStatusCallback_ = std::shared_ptr<ClientResumeStatusCallback>(std::move(statusCallback));
```

Comment:
I would also not allow server RS instance to execute this method. Also not resumable RS should fail in this call.
(CHECK)